app/controllers/microposts_controller.rb

```diff
@@ -5,27 +5,43 @@ class MicropostsController < ApplicationController
   before_action :logged_in_user, only: %i[create destroy]
   before_action :correct_user, only: :destroy
   def create
-    @micropost = current_user.microposts.build(micropost_params)
+    if params[:micropost][:micropost_id]
+      micropost = Micropost.find_by(id: params[:micropost][:micropost_id])
+      @micropost = micropost.microposts.build(micropost_params)
+    else
+      @micropost = current_user.microposts.build(micropost_params)
+    end
     @micropost.image.attach(params[:micropost][:image])
     if @micropost.save
       flash[:success] = 'Micropost created!'
       redirect_to root_url
     else
-      @feed_items = current_user.feed.paginate(page: params[:page])
+      @feed_items = current_user.feed.where(micropost_id: nil).paginate(page: params[:page])
       render 'static_pages/home'
     end
   end
 
   def destroy
-    @micropost.destroy
+    delete_cmt(@micropost.id)
```

Comment:
Use dependent in model relation
example:
`has_many :microposts, class_name: 'Micropost', foreign_key: 'parent_id', dependent: :destroy`